### PR TITLE
Removing last node from main graph makes execution error

### LINKED ISSUE #9172: Removing last node from main graph makes execution error
As in description: the error appears:

image

And then nothing can be done in the graph.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #9172

Changelog:
- update: insert `Empty` IR node for the empty method definition bodies
- update: generate node resulting in `Nothing` for `Empty` IR nodes
- refactor: remove redundant checked exceptions in `EnsoParserTest`

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

File: engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/optimise/LambdaConsolidate.scala [hunk 1]
```diff
@@ -131,7 +131,7 @@ case object LambdaConsolidate extends IRPass {
     * @return the optimised version of `function`, with any directly chained
     *         lambdas collapsed
     */
-  def collapseFunction(
+  private def collapseFunction(
```
Comment (JaroslavTulach) [inline engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/optimise/LambdaConsolidate.scala:134]: `private` is good!

File: engine/runtime-integration-tests/src/test/java/org/enso/compiler/ErrorCompilerTest.java [hunk 2]
```diff
@@ -500,10 +502,10 @@ public void testNPE183863754() throws Exception {
          42
     """);
     var errors = ir.preorder().filter(Syntax.class::isInstance).map(Syntax.class::cast);
-    assertEquals("Two errors", 2, errors.size());
-    assertTrue(errors.head().reason() instanceof Syntax.UnsupportedSyntax);
-    assertEquals(errors.head().location().get().start(), 0);
-    assertEquals(errors.head().location().get().length(), 6);
+    assertEquals("One error", 1, errors.size());
+    assertEquals(Syntax.UnexpectedExpression$.MODULE$, errors.head().reason());
```
Comment (JaroslavTulach) [inline engine/runtime-integration-tests/src/test/java/org/enso/compiler/ErrorCompilerTest.java:506]: Makes sense. One error is still reported and that should be enough to warn the programmer that something is wrong in this case.

File: engine/runtime-parser/src/test/java/org/enso/compiler/core/EnsoParserTest.java [hunk 3]
```diff
@@ -447,7 +447,7 @@ public void testAssignments() throws Exception {
   }
 
   @Test
-  public void testNumberTimes() throws Exception {
+  public void testNumberTimes() {
```
Comment (JaroslavTulach) [inline engine/runtime-parser/src/test/java/org/enso/compiler/core/EnsoParserTest.java:450]: Can you avoid changing unrelated code?

**Review by JaroslavTulach — APPROVED** [review]

File: engine/runtime-parser/src/test/java/org/enso/compiler/core/EnsoParserTest.java (hunk 3 above)
Comment (4e6) [reply engine/runtime-parser/src/test/java/org/enso/compiler/core/EnsoParserTest.java:450]: I follow [Boy Scout Rule](https://wiki.c2.com/?BoyScoutRule)
